### PR TITLE
过去一下代码

### PR DESCRIPTION
**Pull request by GameKeys**
Re: cocos-creator/fireball#

Changes proposed in this pull request:
 * 

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [ ] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [ ] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [ ] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [ ] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
  - [ ] Make sure any runtime log information in `cc.log` , `cc.error`, `cc.warn` or `cc.assert` has been moved into `DebugInfos.js` with an ID

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/value-types/CCRect.js [hunk 1]
```diff
@@ -221,6 +221,19 @@ Object.defineProperty(proto, 'size', {
         this.height = value.height;
     }
 });
+/**
+ * @property origin
```
Comment (jareguo) [inline cocos2d/core/value-types/CCRect.js:225]: 去掉 `@property origin`，保留 `@property {Vec2} origin` 就行

File: cocos2d/core/value-types/CCRect.js [hunk 2]
```diff
@@ -221,6 +221,19 @@ Object.defineProperty(proto, 'size', {
         this.height = value.height;
     }
 });
+/**
```
Comment (jareguo) [inline cocos2d/core/value-types/CCRect.js:224]: 上面空一行

**Review by jareguo — CHANGES_REQUESTED** [review]

**Review by jareguo — CHANGES_REQUESTED** [review]
你的仓库已经太久没更新了，请拉取我们最新的代码，做一次 rebase 操作再提交
image

File: cocos2d/core/CCGame.js [hunk 3]
```diff
@@ -356,31 +360,51 @@ var game = /** @lends cc.game# */{
...
                 }
                 this._persistRootNodes[id] = node;
                 node._persistNode = true;
+                return id;
             }
         }
     },
+    getPersistRootNode:function(key){
+         if (!key) {
+            cc.warn(' getPersistRootNode key==null');
```
Comment (jareguo) [inline cocos2d/core/CCGame.js]: Make sure any runtime log information in cc.log , cc.error, cc.warn or cc.assert has been moved into DebugInfos.js with an ID

File: cocos2d/core/CCGame.js [hunk 4]
```diff
@@ -341,13 +341,17 @@ var game = /** @lends cc.game# */{
      * The target node must be placed in the root level of hierarchy, otherwise this API won't have any effect.
      * @method addPersistRootNode
      * @param {Node} node - The node to be made persistent
+     * @return {key} 返回创建使用的key -1失败 其它成功
      */
     addPersistRootNode: function (node) {
-        if (!(node instanceof cc.Node) || !node.uuid) {
+        return this.setPersistRootNode(node.uuid,node);
+    },
+    setPersistRootNode: function (key,node) {
```
Comment (jareguo) [inline cocos2d/core/CCGame.js]: 请补充 API 注释

File: cocos2d/core/CCGame.js [hunk 5]
```diff
@@ -341,13 +341,17 @@ var game = /** @lends cc.game# */{
      * The target node must be placed in the root level of hierarchy, otherwise this API won't have any effect.
      * @method addPersistRootNode
      * @param {Node} node - The node to be made persistent
+     * @return {key} 返回创建使用的key -1失败 其它成功
```
Comment (jareguo) [inline cocos2d/core/CCGame.js]: `@return {String} key`
另外请用英文注释，注意空格

File: cocos2d/core/CCGame.js [hunk 6]
```diff
@@ -356,31 +360,51 @@ var game = /** @lends cc.game# */{
...
-                    return;
+                    return -1;
                 }
                 this._persistRootNodes[id] = node;
                 node._persistNode = true;
+                return id;
             }
         }
     },
+    getPersistRootNode:function(key){
```
Comment (jareguo) [inline cocos2d/core/CCGame.js]: 请补充 API 注释

File: cocos2d/core/CCGame.js [hunk 7]
```diff
@@ -356,31 +360,51 @@ var game = /** @lends cc.game# */{
...
+        return this._persistRootNodes[key];
+    },
+
 
     /**
      * Remove a persistent root node
      * @method removePersistRootNode
-     * @param {Node} node - The node to be removed from persistent node list
+     * @param {Node|Number} nodeOrKey - The node to be removed from persistent node list
+     * @return {Boolean} false 失败 true 成功
```
Comment (jareguo) [inline cocos2d/core/CCGame.js]: 请不要在 API 中出现中文

File: cocos2d/core/CCGame.js [hunk 8]
```diff
@@ -356,31 +360,51 @@ var game = /** @lends cc.game# */{
...
      */
-    removePersistRootNode: function (node) {
-        if (node !== this._ignoreRemovePersistNode) {
-            var id = node.uuid || '';
-            if (node === this._persistRootNodes[id]) {
+    removePersistRootNode: function (nodeOrKey) {
+        if (nodeOrKey !== this._ignoreRemovePersistNode) {
+             var id =''
+            if(nodeOrKey instanceof cc.Node){
```
Comment (jareguo) [inline cocos2d/core/CCGame.js:405]: 注意空格：Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline

File: cocos2d/core/CCGame.js (hunk 5 above)
Comment (jareguo) [reply cocos2d/core/CCGame.js]: 为什么要有一个 KEY？

File: cocos2d/core/CCGame.js [hunk 9]
```diff
@@ -356,31 +360,51 @@ var game = /** @lends cc.game# */{
...
+        }
+        return this._persistRootNodes[key];
+    },
+
 
     /**
      * Remove a persistent root node
      * @method removePersistRootNode
-     * @param {Node} node - The node to be removed from persistent node list
+     * @param {Node|Number} nodeOrKey - The node to be removed from persistent node list
```
Comment (jareguo) [inline cocos2d/core/CCGame.js]: node.uuid 是 String，不是 Number

File: cocos2d/core/CCGame.js [hunk 10]
```diff
@@ -356,31 +360,51 @@ var game = /** @lends cc.game# */{
...
-    removePersistRootNode: function (node) {
-        if (node !== this._ignoreRemovePersistNode) {
-            var id = node.uuid || '';
-            if (node === this._persistRootNodes[id]) {
+    removePersistRootNode: function (nodeOrKey) {
+        if (nodeOrKey !== this._ignoreRemovePersistNode) {
+             var id =''
+            if(nodeOrKey instanceof cc.Node){
+                id = nodeOrKey.uuid || '';
+            }else if(typeof(nodeOrKey) == 'number'){
```
Comment (jareguo) [inline cocos2d/core/CCGame.js]: node.uuid 是 String，不是 Number

File: cocos2d/core/CCGame.js [hunk 11]
```diff
@@ -356,31 +360,51 @@ var game = /** @lends cc.game# */{
...
-     * @param {Node} node - The node to be removed from persistent node list
+     * @param {Node|Number} nodeOrKey - The node to be removed from persistent node list
+     * @return {Boolean} false 失败 true 成功
      */
-    removePersistRootNode: function (node) {
-        if (node !== this._ignoreRemovePersistNode) {
-            var id = node.uuid || '';
-            if (node === this._persistRootNodes[id]) {
+    removePersistRootNode: function (nodeOrKey) {
+        if (nodeOrKey !== this._ignoreRemovePersistNode) {
```
Comment (jareguo) [inline cocos2d/core/CCGame.js:403]: 这个 if 判断仅在 nodeOrKey instanceof cc.Node 时有效，放在这里不合适

**Review by jareguo — CHANGES_REQUESTED** [review]

File: cocos2d/core/CCGame.js (hunk 5 above)
Comment (GameKeys) [reply cocos2d/core/CCGame.js]: 就是不知道怎么获取常驻节点！才加key

**Comment by jareguo** [comment]
感觉好像没必要给常驻节点加 key，因为常驻节点本来就能用 cc.find 之类的 API 查到啊？或者你调用 cc.game.addPersistRootNode 时，就能顺便存到你的变量里了？

**Comment by jareguo** [comment]
太久没下文，本 PR 关闭，部分代码已人工摘录到 #1617，谢谢！

**Pull request closed without merging**